### ISSUE TITLE
Build Error: import "unix-bytestring" System.Posix.IO.ByteString (fdPreadBuf)

### ISSUE THREAD

**Issue by jritchie-nullable** [input-output-hk/ouroboros-network#4471]
I apologize if this issue is a bit lacking in details.  I have little/no experience with Cabal / Haskell / Cardano.  I believe this is the correct repo for this issue.

In trying to build cardano-node it seems that unix-bytestring v0.4.0 has renamed System.Posix.IO.ByteString.

Chunk of log
```
[ 24 of 214] Compiling Ouroboros.Consensus.Storage.IO ( src-unix/Ouroboros/Consensus/Storage/IO.hs, dist/build/Ouroboros/Consensus/Storage/IO.o, dist/build/Ouroboros/Consensus/Storage/IO.dyn_o )

src-unix/Ouroboros/Consensus/Storage/IO.hs:28:1: error:
    Could not find module ‘System.Posix.IO.ByteString’
    Perhaps you meant
      System.Posix.IO.ByteString (from unix-2.7.2.2)
      System.Posix.ByteString (from unix-2.7.2.2)
      System.Posix.Env.ByteString (from unix-2.7.2.2)
   |
28 | import           "unix-bytestring" System.Posix.IO.ByteString (fdPreadBuf)

```
updating cabal.project.local with a package constraint seems to resolve the problem.

`constraints: unix-bytestring == 0.3.7.8`

**Comment by Kriator7** [comment]
System.Posix.IO.ByteString cannot be found, which means that it is not installed or not properly configured.

**Comment by Kriator7** [comment]
sudo apt-get install libgmp-dev zlib1g-dev libtinfo-dev

cabal update

cabal install unix-bytestring

cabal build all


Let me know if this helps..

**Comment by jritchie-nullable** [comment]
@Kriator7 I've hunted down the confusion.  Cardano-node's latest release is pinned to ouroboros-consensus == 0.1.0.3 which utilized the unix-bytestring package (It was replaced in this commit: [8dab74ff](https://github.com/input-output-hk/ouroboros-network/commit/8dab74ff3cd70c5bbf0ac7a4de4b2ed116871980)).  unix-bytestring's latest release 0.4.0 has the breaking change which causes ouroboros-consensus' 0.1.0.3 build to fail.  I suppose this will be resolved in the next? release of ouroboros-consensus.

This issue might belong better in cardano-node.  

Thanks!
